main.go

```diff
@@ -0,0 +1,282 @@
+	"fmt"
+	"iter"
+	"os"
+	"slices"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/service/budgets"
+	"github.com/aws/aws-sdk-go-v2/service/budgets/types"
+	policyManager "github.com/compliance-framework/agent/policy-manager"
+	"github.com/compliance-framework/agent/runner"
+	"github.com/compliance-framework/agent/runner/proto"
+	"github.com/hashicorp/go-hclog"
+	goplugin "github.com/hashicorp/go-plugin"
+)
+
+type CompliancePlugin struct {
+	logger hclog.Logger
+	config map[string]string
+}
+
+func (l *CompliancePlugin) Configure(req *proto.ConfigureRequest) (*proto.ConfigureResponse, error) {
+	l.config = req.GetConfig()
+	return &proto.ConfigureResponse{}, nil
+}
+
+func (l *CompliancePlugin) Eval(request *proto.EvalRequest, apiHelper runner.ApiHelper) (*proto.EvalResponse, error) {
+	ctx := context.TODO()
+	evalStatus := proto.ExecutionStatus_SUCCESS
+	var accumulatedErrors error
+
+	l.logger.Info("Starting eval!")
+
+	activities := make([]*proto.Activity, 0)
+	activities = append(activities, &proto.Activity{
+		Title:       "Collect AWS Budget configurations",
+		Description: "Using the Golang AWS SDK, fetch all Budget information and metadata.",
+		Steps: []*proto.Step{
+			{
+				Title:       "Configure AWS Client",
+				Description: "Using the default configuration loaders, create a AWS client for querying the AWS API",
+			},
+			{
+				Title:       "Create a new AWS Budgets Client",
+				Description: "In order to describe active budgets, we instantiate a new AWS Budgets client using the Golang AWS SDK.",
+			},
+			{
+				Title:       "Describe Budgets",
+				Description: "Using the newly constructed AWS client, Describe all created budgets, and store them in local memory.",
+			},
+		},
+	})
+
+	cfg, err := config.LoadDefaultConfig(ctx, config.WithRegion(os.Getenv("AWS_REGION")))
+	if err != nil {
+		l.logger.Error("unable to load SDK config", "error", err)
+		evalStatus = proto.ExecutionStatus_FAILURE
+		accumulatedErrors = errors.Join(accumulatedErrors, err)
+	}
+
+	budgetsClient := budgets.NewFromConfig(cfg)
+
+	accountId := l.config["account_id"]
+
+
+
+	// Run policy checks
+	for budget, err := range getBudgets(ctx, budgetsClient, &accountId) {
+		l.logger.Debug(fmt.Sprintf("Found budget: %v", aws.ToString(budget.BudgetName)))
+		if err != nil {
+			l.logger.Error("unable to get budget", "error", err)
+			evalStatus = proto.ExecutionStatus_FAILURE
+			accumulatedErrors = errors.Join(accumulatedErrors, err)
+			break
+		}
+
+		alertCount := 0
+
+		for _, err := range getNotificationsForBudget(ctx, budgetsClient, &accountId, budget.BudgetName) {
+			if err != nil {
+				l.logger.Error("unable to get notification", "error", err)
+				evalStatus = proto.ExecutionStatus_FAILURE
+				accumulatedErrors = errors.Join(accumulatedErrors, err)
+				break
+			}
+			alertCount += 1
+		}
+
+		labels := map[string]string{
+			"provider":    "aws", 
+			"type":        "budget",
```

Comment:
Good to add other metadata information:
```suggestion
			"type":        "budget",
			"account-id:   &accountId,
```